tensorflow/core/grappler/optimizers/constant_folding.cc

```diff
@@ -2382,20 +2382,34 @@ bool ConstantFolding::SimplifySwitch(GraphDef* optimized_graph, NodeDef* node) {
   return false;
 }
 
-bool ConstantFolding::IsReductionCandidateForSimplification(
-    const NodeDef& node, const GraphProperties& properties,
-    TensorShapeProto* input_tensor_shape, TensorShapeProto* output_tensor_shape,
-    bool* is_single_element_op) const {
+bool ConstantFolding::IsReductionWithConstantIndices(
+    const NodeDef& node, bool* indices_is_empty) const {
   // Ensure its an appropriate Reduce node.
   if (!IsReduction(node) || node.input_size() < 2) {
     return false;
   }
   // Ensure that the axes to reduce by are constant.
   NodeDef* reductions_indices = node_map_->GetNode(node.input(1));
-  if (!IsReallyConstant(*reductions_indices)) {
+  if (!IsReallyConstant(*reductions_indices) ||
+      !reductions_indices->attr().count("value")) {
     return false;
   }
+  const TensorProto& reduction_indices_tensor =
+      reductions_indices->attr().at("value").tensor();
+  *indices_is_empty = false;
+  for (const auto& dim : reduction_indices_tensor.tensor_shape().dim()) {
```

Comment:
Would it be better to use TensorShapeUtils here?

Reply:
Thanks for the suggestion. I've replaced it with `TensorShape(reduction_indices_shape).num_elements()`.